Microsoft.Toolkit.Uwp/Helpers/DispatcherHelper.cs

```diff
@@ -22,6 +22,7 @@ public static class DispatcherHelper
         /// <param name="priority">Dispatcher execution priority, default is normal.</param>
         /// <returns>An awaitable <see cref="Task"/> for the operation.</returns>
         /// <remarks>If the current thread has UI access, <paramref name="function"/> will be invoked directly.</remarks>
+        [Obsolete("This method should be replaced with CoreApplication.MainView.DispatcherQueue.EnqueueAsync(function, priority).")]
```

Comment:
This suggestion is tricky... This will work fine for now, but in the future (v8, WinUI3), when an app is running as a Win32 app on WinUI3 Desktop, CoreApplication.MainView will be null. Calling DispatcherQueue.GetForCurrentThread() from the current UI thread is still the best approach. I would also update the comments bellow mentioning that the call to DispatcherQueue.GetForCurrentThread() should be made from the current UI thread, maybe even pointing to the documentation. I know you did that on the ObsoleteAttribute for this class, but I can see the "bug" reports showing up already if we don't make this transition very smooth.

Reply:
Right, that's a good point. I was mostly just assuming that by the time WinUI 3 would be out, this class would've been removed completely already. Just to be clear, since using these extensions if you're _already_ on the UI thread is basically useless and only slows things down, at this point should we also mention that the target `DidpatcherQueue` should be retrieved when on the UI thread and then stored somewhere (eg. a field) so that the extension can then be invoked on it when _not_ running on that thread? 🤔

I mean invoking these extensions directly on a `DispatcherQueue.GetForCurrentThread()` should basically never be used, right?

Reply:
@azchohfi thoughts on what we should do here. I do see that `DispatcherHelper` will disappear in '8.0', but do want to make sure we setup a better transition for code for folks so they don't have to migrate now and then migrate again.

Reply:
@michael-hawker Updated the text as suggested in https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3498/commits/9ada6d9f385905a34c08e50140cd7c4c838120aa 👍